src/robotFactory.js

```diff
@@ -1,10 +1,70 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = Object.keys(coords).length ? coords : {
+      x: 0, y: 0,
+    };
```

Comment:
Here's good way to check each coordinate separately.
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```